Examples/draggable-graph.lisp

```diff
@@ -85,42 +75,39 @@
     (make-rectangle* x0 y0 x1 y1)))
 
 (define-draggable-graph-demo-command (com-drag-node)
-    ((record t) (x real) (y real))
-  (let* ((graph-node (find-graph-node record))
-	 (edges (node-edges graph-node))
-	 (erase-region (stupid-copy-rectangle
-                        (reduce (lambda (x &optional y)
-                                  (if y (region-union x y) x))
-                                edges))))
-    (multiple-value-bind (px py) (output-record-position graph-node)
-      (let ((graph (output-record-parent graph-node))
-	    (x-offset (- x px))
-	    (y-offset (- y py)))
-	(assert (typep graph 'graph-output-record))
-	(erase-output-record graph-node *standard-output*)
-	(dolist (edge edges)
-	  (clear-output-record edge))
-	(when edges (repaint-sheet *standard-output* erase-region))
-	(multiple-value-bind (final-x final-y)
-	    (drag-output-record *standard-output* graph-node
-				:erase-final t
-				:finish-on-release t)
-	  (setf (output-record-position graph-node)
-		(values (- final-x x-offset) (- final-y y-offset)))
-          
-	  (add-output-record graph-node graph)
-	  (redisplay-edges graph edges)
-	  (repaint-sheet *standard-output* graph-node))))))
-         
+    ((record t) (offset-x real :default 0) (offset-y real :default 0))
+  (let* ((stream *standard-output*)
+         (node-record (find-graph-node record))
+         (edge-records (node-edges node-record))
+         (graph-record (output-record-parent node-record))
+         (erase-region))
+    (assert (typep graph-record 'graph-output-record))
+    (erase-output-record node-record stream)
+    (drag-output-record
+     stream node-record
+     :feedback (lambda (record stream old-x old-y x y mode)
```

Comment:
please put `stream` in the same line as the function. In general please avoid style, where first function argument gets its own line. I don't recall this style being used anywhere in the source code and I'm not a fan of it myself. If lambda function is nested too deep for feedback use flet.